Issue #2231 Replaced invalid remove() calls with pop(,None) for Python dictionaries

UO-BS opened the pull request:
This is in relation to Issue #2231 

When converting a Java map to a Python dictionary, the remove(arg) method of maps must be converted to its equivalent pop(arg,None) for dictionaries.


## Changes

Appended a new rule to the TXL files to do the above. 
Repaired a bad test case which used the incorrect remove(arg) method.

UO-BS committed:
Replaced invalid remove() calls with pop(,None) for JavaToPython

Added a new rule to the TXL files so that any java map that uses the remove() method will be converted to a python dict that uses the pop(,None) method.

Previously it called remove() on a python dict, which does not compile.

TimLethbridge commented:
We note that this works for command line work: the UmpleToPython/txl/umpleJavaToPython.ctxl is being built. The command line versions work

But the copy of that file in umpleonline/scripts/txl/umpleJavaToPython.ctxl is not being made.

Actually it is a very stinky smell that there is a dupliucate file.

The solution is as follows: take umpleonline/scripts/txl/umpleJavaToPython.ctxl  out of version control. The modify the build script that does packageUmpleOnline to copy the file every time it is run. (that script in ant is build.umple.xml line 461 (look for the related place in Gradle)

TimLethbridge committed:
Merge branch 'master' into issue2231

TimLethbridge committed:
Merge branch 'master' into issue2231

UO-BS committed:
The JavaToPython ctxl file will now be copied to umpleonline/scripts

Stopped git tracking the second copy of the umpleJavaToPython.ctxl file.

This file will be copied from the UmpleToPython whenever the user calls the task "packageUmpleonline" in their build.

UO-BS committed:
Made gradle task description more clear
